src/analysis/function_parameters.rs

```diff
@@ -417,5 +426,5 @@ fn has_length(env: &Env, typ: TypeId) -> bool {
 }
 
 pub fn async_param_to_remove(name: &str) -> bool {
-    name == "user_data"
+    name == "user_data" || name.ends_with("data")
```

Comment:
Can be done based on the attributes (`closure` attribute) instead of guesswork based on parameter names

Reply:
Again, async fix. I don't intend to fix this code (the async one) in this PR.

Reply:
I assumed that this is also called from your new code now because you changed this function. Why was that change needed?

Reply:
A function was newly generated because the closure type could be generated. But it'll need to be fixed \*soon\*.